corehq/apps/userreports/reports/filters/choice_providers.py

```diff
@@ -394,6 +397,42 @@ def default_value(self, user):
         return None
 
 
+class DomainChoiceProvider(ChainableChoiceProvider):
+
+    def _query_domains(self, domain, query_text):
+        registry_slug = self.report.config.registry_slug
+        try:
+            registry = DataRegistry.objects.accessible_to_domain(domain, slug=registry_slug).get()
+        except DataRegistry.DoesNotExist:
+            return {domain}
+        domains = {domain}
+        domains.update(registry.get_granted_domains(domain))
+        if query_text:
+            domains = {domain for domain in domains if re.search(query_text, domain)}
+        return list(domains)
+
+    def query(self, query_context):
+        domains = self._query_domains(self.domain, query_context.query)
+        domains.sort()
+        return self._domains_to_choices(
+            domains[query_context.offset:query_context.offset + query_context.limit]
+        )
+
+    def query_count(self, query):
+        return len(self._query_domains(self.domain, query))
+
+    def get_choices_for_known_values(self, values, user):
+        domains = self._query_domains(self.domain, None)
+        domain_options = [domain for domain in domains if domain in values]
+        return self._domains_to_choices(domain_options)
+
+    def default_value(self, user):
+        return self._domains_to_choices([self.domain])
+
+    def _domains_to_choices(self, domains):
```

Comment:
you could memoize this too

Reply:
ran into a `TypeError: unhashable type: ‘list’` when trying for this one